server/processors/src/test/java/org/hawkular/apm/server/processor/nodedetails/NodeDetailsDeriverTest.java

```diff
@@ -137,4 +141,31 @@ public void testProcessCommonProperties() throws RetryAttemptException {
         assertEquals(new HashSet<>(Arrays.asList(p1,p2,p3,p4)), details.get(2).getProperties());
     }
 
+    @Test
+    public void testProcessInternalFragmentWithProducer() {
```

Comment:
Maybe rename it to `testInitial` or something like that? As it is testing only that.